packages/gafl-webapp-service/src/pages/contact/contact/update-transaction.js

```diff
@@ -9,32 +9,41 @@ export default async request => {
 
   licensee.preferredMethodOfNewsletter = licensee.preferredMethodOfNewsletter || HOW_CONTACTED.none
 
-  switch (payload['how-contacted']) {
-    case 'email':
-      licensee.preferredMethodOfConfirmation = HOW_CONTACTED.email
-      licensee.preferredMethodOfReminder = HOW_CONTACTED.email
-      licensee.email = payload.email
-      licensee.mobilePhone = null
-      break
+  if (isPhysical(permission)) {
+    switch (payload['how-contacted']) {
+      case 'email':
+        licensee.preferredMethodOfReminder = HOW_CONTACTED.email
+        licensee.email = payload.email
+        break
+      case 'text':
+        licensee.preferredMethodOfReminder = HOW_CONTACTED.text
+        licensee.mobilePhone = payload.text
+        break
+      default:
+        licensee.preferredMethodOfReminder = HOW_CONTACTED.letter
+    }
+  } else {
+    switch (payload['how-contacted']) {
+      case 'email':
+        licensee.preferredMethodOfConfirmation = HOW_CONTACTED.email
+        licensee.preferredMethodOfReminder = HOW_CONTACTED.email
+        licensee.email = payload.email
+        licensee.mobilePhone = null
+        break
 
-    case 'text':
-      licensee.preferredMethodOfConfirmation = HOW_CONTACTED.text
-      licensee.preferredMethodOfReminder = HOW_CONTACTED.text
-      licensee.mobilePhone = payload.text
-      licensee.email = licensee.preferredMethodOfNewsletter === HOW_CONTACTED.email ? licensee.email : null
-      break
+      case 'text':
+        licensee.preferredMethodOfConfirmation = HOW_CONTACTED.text
+        licensee.preferredMethodOfReminder = HOW_CONTACTED.text
+        licensee.mobilePhone = payload.text
+        licensee.email = licensee.preferredMethodOfNewsletter === HOW_CONTACTED.email ? licensee.email : null
+        break
 
-    default:
-      if (isPhysical(permission)) {
-        licensee.preferredMethodOfConfirmation = HOW_CONTACTED.letter
-        licensee.preferredMethodOfReminder = HOW_CONTACTED.letter
-      } else {
+      default:
         licensee.preferredMethodOfConfirmation = HOW_CONTACTED.none
         licensee.preferredMethodOfReminder = HOW_CONTACTED.none
-      }
-
-      licensee.mobilePhone = null
-      licensee.email = licensee.preferredMethodOfNewsletter === HOW_CONTACTED.email ? licensee.email : null
+        licensee.mobilePhone = null
+        licensee.email = licensee.preferredMethodOfNewsletter === HOW_CONTACTED.email ? licensee.email : null
+    }
```

Comment:
This is starting to feel overly complex - switches inside of if...else. Please could you extract, perhaps to `getPhysicalConfirmationsAndReminders` and `getDigitalConfirmationsAndReminders` methods (unless you can think of a better way)?

Reply:
Yes I will extract it out into separate methods.